src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -258,15 +258,35 @@ public Screen getNextScreen(boolean needsFullEntityScreen, EntityScreenContext e
             queryScreen.init(sessionWrapper);
             return queryScreen;
         } else if (next.equalsIgnoreCase(SessionFrame.STATE_SYNC_REQUEST)) {
-            String username = session.getAsUser() != null ?
-                    StringUtils.getFullUsername(session.getAsUser(), session.getDomain()) : null;
-            FormplayerSyncScreen syncScreen = new FormplayerSyncScreen(username);
-            syncScreen.init(sessionWrapper);
-            return syncScreen;
+            return getSyncScreen();
         }
         throw new RuntimeException("Unexpected Frame Request: " + sessionWrapper.getNeededData());
     }
 
+    public Screen getNextScreenIfSyncScreen(boolean needsFullEntityScreen, EntityScreenContext entityScreenContext) throws CommCareSessionException {
```

Comment:
This could do with a doc string giving more context. I don't quite follow what it's trying to do. It seems like it's looking for a sync screen anywhere in the stack up to that point. Is that correct?

Reply:
It's what `getNextScreen` does but only initializes and returns the screen if it's a `syncScreen`. 

What do you think of this docstring?
```
Get next screen for the current request, based on the current state of the session, 
but only initialize and return the screen if it is of type `FormplayerSyncScreen`.     ```

Reply:
[9580cb2](https://github.com/dimagi/formplayer/pull/1608/commits/9580cb245cc2008d1284d99b0ac32386263d3a83)